SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -295,47 +292,29 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 	Connection->OnConnectedCallback.BindUObject(this, &USpatialNetDriver::OnConnectionToSpatialOSSucceeded);
 	Connection->OnFailedToConnectCallback.BindUObject(this, &USpatialNetDriver::OnConnectionToSpatialOSFailed);
 
-	bool bUseReceptionist = true;
-	bool bShouldLoadFromURL = true;
-
 	// If this is the first connection try using the command line arguments to setup the config objects.
 	// If arguments can not be found we will use the regular flow of loading from the input URL.
 	if (!GameInstance->GetFirstConnectionToSpatialOSAttempted())
 	{
-		bool bSuccessfullyLoadedFromCommandLine;
-		StartSetupConnectionConfigFromCommandLine(bSuccessfullyLoadedFromCommandLine, bUseReceptionist);
-		bShouldLoadFromURL = !bSuccessfullyLoadedFromCommandLine;
 		GameInstance->SetFirstConnectionToSpatialOSAttempted();
+		if (!TrySetupConnectionConfigFromCommandLine())
+		{
+			SetupConnectionConfigFromURL(URL);
+		}
 	}
 	else if (URL.Host == SpatialConstants::RECONNECT_USING_COMMANDLINE_ARGUMENTS)
 	{
-		bool bSuccessfullyLoadedFromCommandLine;
-		StartSetupConnectionConfigFromCommandLine(bSuccessfullyLoadedFromCommandLine, bUseReceptionist);
-
-		if (!bSuccessfullyLoadedFromCommandLine)
+		if (TrySetupConnectionConfigFromCommandLine())
 		{
-			if (bUseReceptionist)
-			{
-				Connection->ReceptionistConfig.LoadDefaults();
-			}
-			else
-			{
-				Connection->LocatorConfig.LoadDefaults();
-			}
+			Connection->SetConnectionType(ESpatialConnectionType::Receptionist);
```

Comment:
Is this line necessary? Will it not have been done as part of TrySetupConnectionConfigFromCommandLine?

Reply:
Never mind this. I think the if statement needs to change in which case this make more sense.